mypy/options.py

```diff
@@ -302,11 +302,13 @@ def clone_for_module(self, module: str) -> 'Options':
 
         return options
 
-    def module_matches_pattern(self, module: str, pattern: Pattern[str]) -> bool:
-        # If the pattern is 'mod.*', we want 'mod' to match that too.
-        # (That's so that a pattern specifying a package also matches
-        # that package's __init__.)
-        return pattern.match(module) is not None or pattern.match(module + '.') is not None
+    def compile_glob(self, s: str) -> Pattern[str]:
+        # Compile one of the glob patterns to a regex so that '.*' can
+        # match *zero or more* module sections. This means we compile
+        # '.*' into '(\..*)?'. We also need to escape .s in the glob, so
+        # we hackily rewrite .s to ,s to accomplish this.
+        s = s.replace('.', ',').replace(',*', '(\..*)?').replace(',', '\.')
```

Comment:
This doesn't work right for *.foo I think. Also maybe the code would be more readable if you just started with module.split('.') and applied re.escape() to parts that aren't `*`, and something else for other parts, special-casing a `*` at the front.